master/internal/task/allocation_service.go

```diff
@@ -140,6 +140,18 @@ func (as *allocationService) SetProxyAddress(
 	return ref.SetProxyAddress(ctx, addr)
 }
 
+// SetAcceleratorData sets the accleration data of the allocation.
+func (as *allocationService) SetAcceleratorData(
+	ctx context.Context,
+	accData model.AcceleratorData,
+) error {
+	ref, err := as.getAllocation(accData.AllocationID)
```

Comment:
why not just have this call `AddAllocationAcceleratorData` directly? i don't think we need the indirection through `ref`. `ref` is really just for the stateful stuff.

Reply:
Done